MIGRATING_TO_NEW_FORM.md

```diff
@@ -0,0 +1,256 @@
+# Migration to New Form
+
+## Background
+
+The form has been updated to add Typescript supporting by wrapping [`react-hook-form`](https://react-hook-form.com).
+
+This is a non breaking change as the previous implementation of Form is still supported for the foreseable future. The new pattern is opt in see Form section below.
+
+**NOTE**: It is strongly recommended to go through react hook form docs and familiarize with patterns using the doc.
+
+## Form
+
+### Using new form
+
+To use the new form pattern, provide the `Form` component with `methods=` prop, which expects the return value from `useBasisForm` hook (see below).
+
+```jsx
+import { useBasisForm, Form } from "basis";
+
+const App = () => {
+  const { methods, ...other } = useBasisForm();
+
+  return (
+    <Form methods={methods} onSubmit={console.log}>
+      {/* Implementation of form*/}
+    </Form>
+  );
+};
+```
+
+### Default values behaviour
+
+The new form no longer needs `initialValues`. It will default the components provided to the relavent empty state. If non empty state is required, they must be passed into `useBasisForm`.
+
+_Note: this will only defalut the feilds provided leaving all others empty._
+
+```jsx
+import { useBasisForm, Form } from "basis";
+
+const App = () => {
+  const { methods, ...other } = useBasisForm({
+    defaultValues: {
+      field1: "non-empty",
+    },
+  });
+  /* rest of form */
+};
+```
+
+### Using with Typescript
+
+To use the Form with Typescript provide the types expected by the return value into the `useBasisForm<Types>` generic.
+
+```tsx
+import { useBasisForm, Form } from "basis";
+
+interface FormState {
+  field1: string;
+  field2: string;
+  fieldDate: {
+    day: string;
+    month: string;
+    year: string;
+  };
+}
+
+const App = () => {
+  const { methods, ...other } = useBasisForm<FormState>();
+
+  return (
+    <Form methods={methods} onSubmit={console.log}>
+      {/* Implementation of form*/}
+    </Form>
+  );
+};
+```
+
+### `onSubmit` changes
+
+onSubmit will only be called with values and will only be called when a successful submission takes place. i.e. after all user validation has passed.
+
+## Field
+
+A new `Field` component has been added as a return value from `useBasisForm`. This component acts as a wrapper for other Form components handling their validation and Form state.
+
+_Note: defaultValue can also be defined on Field_
+
+A basic Form would look like the following:
+
+```jsx
+import { useBasisForm, Form, Input, DateInput } from "basis";
+
+const App = () => {
+  const { methods, Field } = useBasisForm();
+
+  const onSubmit = (values) => {
+    console.log(values);
+  };
+
+  return (
+    <Form methods={methods}>
+      <Field name="firstName" label="First Name" as={Input} />
```

Comment:
Let's put sentence case labels everywhere to encourage people to do it right.
First Name => First name